docker/ci/build.ps1

```diff
@@ -0,0 +1,106 @@
+$ErrorActionPreference = 'Stop'
+
+function Get-CurrentDirectory
+{
+  $thisName = $MyInvocation.MyCommand.Name
+  [IO.Path]::GetDirectoryName((Get-Content function:$thisName).File)
+}
+
+function Get-ContainerVersion
+{
+  # shallow repositories need to pull remaining code to `git describe` correctly
+  if (Test-Path "$(git rev-parse --git-dir)/shallow")
+  {
+    git fetch --unshallow
+  }
+
+  # tags required for versioning
+  git fetch origin 'refs/tags/*:refs/tags/*'
+  (git describe) -replace '-.*', ''
+}
+
+# installs gems for build and test and grabs base images
+function Invoke-ContainerBuildSetup
+{
+  Push-Location (Get-CurrentDirectory)
+  docker pull ubuntu:16.04
```

Comment:
@underscorgan is this the same thing we figured out we could get rid of by using the `--pull` flag on our docker invocations?

Reply:
Ah yeah, good point. I basically ported over the `Makefile`, but might have been working off an older version. If we can remove it, that'd be great.

Reply:
yeah, we can `docker build --pull` for the same functionality